packages/fast-components/src/text-field/text-field.ts

```diff
@@ -0,0 +1,95 @@
+import { attr, FastElement, observable } from "@microsoft/fast-element";
+
+export enum TextFieldAppearance {
+    filled = "filled",
+    outline = "outline",
+}
+
+export enum TextFieldType {
+    email = "email",
+    password = "password",
+    tel = "tel",
+    text = "text",
+    url = "url",
+}
+
+export class TextField extends FastElement {
+    @attr
+    public appearance: TextFieldAppearance = TextFieldAppearance.outline;
+    private appearanceChanged(): void {
+        this.appearance === TextFieldAppearance.filled
```

Comment:
This is a perfect use for host directives...but I think there would be negative side-effects with my current implementation as it would blow away any end-user classes, when in reality we want to merge them. I think after I get some of my binding refactoring done, we may able to look at improving this so it could be done declaratively instead. No need to change anything now.